### PR TITLE
Add JSON output to `iotedge check`

### PR DESCRIPTION
**Pull request by arsing**
This adds a `--output json` parameter to `iotedge check` that prints
the check results as machine-readable JSON. The file also contains
the host OS name and bitness.

---

```sh
$ iotedge check --help

       -o, --output <FORMAT>                        Output format. [default: text]  [possible values: json, text]


$ iotedge check ... --output json | jq

{
  "os_info": {
    "id": "opensuse-tumbleweed",
    "version_id": "20190325",
    "bitness": 64
  },
  "checks": {
    "certificates-expiry": {
      "result": "Ok"
    },
    "certificates-quickstart": {
      "result": "Warning",
      "details": [
        "Device is using self-signed, automatically generated certs.\nPlease see https://aka.ms/iotedge-prod-checklist-certs for best practices."
      ]
    },
    "config-yaml-well-formed": {
      "result": "Ok"
    },

   ...
```

### CODE REVIEW

**Comment by myagley** [comment]
Should we look at adding an `-o / --output` parameter that changes how the results are written to `stdout` instead of an explicit `--json` parameter that writes to a file? This would then give maximum flexibility.

For example, the current output is:

```
$ iotedge check -o table
```

For json output,

```
$ iotedge check -o json
```

To store to a file:
```
$ iotedge check -o json > check.json
```

This is then consistent with the `az` cli tool as well, which allows different output formatting.

**Comment by arsing** [comment]
I'd proposed `--output json` in the spec meeting but I believe the consensus had been that users would want to be able to see the regular output at the same time.

**Comment by veyalla** [comment]
Can we have both?

**Comment by arsing** [comment]
@veyalla What do you mean by "both" ?

**Comment by arsing** [comment]
(The Windows CI failure appears to be because at least one of the pool.ntp.org servers violates the SNTP RFC. I'll put up a workaround in a separate PR.)

**Comment by veyalla** [comment]
Discussed offline - I meant supporting both `--output json` and `-o json`.

**Comment by arsing** [comment]
@myagley I discussed with Venkat. The JSON output is primarily meant for users to copy-paste into Github issues. Unlike `az` it's not very human-readable - it's not even pretty-printed (which is why my example pipes it through jq) and it's at least two kilobytes long even without whitespace. Pretty-printing the JSON produces many pagefuls of output, that aren't going to be as useful as the regular output.

**Comment by arsing** [comment]
Changed `--json PATH` to `--output json` and made it write to stdout.

File: edgelet/iotedge/src/main.rs [hunk 1]
```diff
@@ -131,6 +131,15 @@ fn run() -> Result<(), Error> {
                         .takes_value(true)
                         .default_value("pool.ntp.org:123"),
                 )
+                .arg(
+                    Arg::with_name("output")
+                        .long("output")
```
Comment (myagley) [inline edgelet/iotedge/src/main.rs]: Can we add a short param `-o`?

Comment (arsing) [reply edgelet/iotedge/src/main.rs]: Done.

**Pull request merged**